partcad-service-json-rpc/src/partcad_service_json_rpc/core/operations.py

```
@@ -0,0 +1,995 @@
+    """
+    pc = session.ensure_partcad()
+    url = params.get("url") or ("file://" + os.getcwd())
+    path = _url_to_path(url)
+    root = os.path.abspath(path)
+    context_id = hashlib.sha256(root.encode("utf-8")).hexdigest()[:16]
+
+    if context_id not in session.contexts:
+        try:
+            session.contexts[context_id] = pc.init(path, user_config=pc.user_config)
+        except (yaml.parser.ParserError, yaml.scanner.ScannerError) as e:
+            raise JsonRpcError(INVALID_CONFIG, "Invalid configuration file", data={"detail": str(e)})
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)